chapters/funkcije/funkcije.tex

```diff
@@ -0,0 +1,226 @@
+\end{examples}
+
+Tu funkcija vrača logično vrednost \verb+bool+, ki je ena izmed \verb+true+ ali
+\verb+false+, uporablja se jo lahko v if stavku. Deluje torej pravilno, saj če
+bo \verb+crka+ enaka a, e, i, o, u, se bo funkcija končala pri return-u v if
+stavku, sicer se bo pa nadaljevala do \verb+return false;+.
+
+
+\section{Spremenljivke in funkcije}
+
+\subsection{Globalne in lokalne spremenljivke}
+
+Spomnimo se, da lahko spremenljivke definiramo na dva načina.
+Tistim, ki so definirane na vrhu, izven funkcije rečemo \emph{globalne
+spremenljivke}, tistim, ki so pa definirane v funkciji pa \emph{lokalne
+spremenljivke}. Zdaj, ko vemo, kaj so funkcije, lahko povemo razliko med temi
+tipi spremenljivk.
+
+Globalne spremenljivke so vidne povsod, torej v vseh funkcijah, medtem ko lahko
+lokalne spremenljivke uporabljamo samo v funkciji, v kateri smo jih definirali.
+
+\begin{examples}
+    %#insert python3 style.py < chapters/funkcije/globalne_lokalne_spremenljivke.cpp
+\end{examples}
+
+Seveda, če globalno spremenljivko nekje spremenimo, se bo spremenila tudi za
+vse druge funkcije.
+
+\subsection{Spreminjanje parametrov}
+
+Če funkciji podamo neko spremenljivko kot
+parameter in poskušamo to spremenljivko v funkciji spreminjati, se obnaša na
+dva različna načina, odvisno od tega, če je spremenljivka seznam.
+
+Če spremenljivka ni seznam (npr. \verb+int+, \verb+char+) se za funkcijo
+ustvari kopija te spremenljivke. Če jo v funkciji spremenimo, to ne bo
+spremenilo izvirne spremenljivke, saj v funkciji delamo s kopijo. Če je seznam,
+bodo pa spremembe na seznam v funkciji vplivale na izvirni seznam, saj se ta ne
+bo prekopiral. V spodnjem primeru sta prikazani obe možnosti.
+
+\begin{examples}
+    %#insert python3 style.py < chapters/funkcije/seznam_se_ne_kopira.cpp
+\end{examples}
+
+\newpage
+
+Sprva ne vidimo smisla, za drugačno obravnavo seznamov, a se izkaže, da je
+lahko ta lastnost tudi uporabna. Če bi želeli napisati funkcijo, ki sprejeme
+seznam in njegovo dolžino, potem pa obrne vrstni red elementov tega seznama,
+lahko to napišemo takole:
+
+\begin{examples}
+
+    %#insert python3 style.py < chapters/funkcije/obrni_seznam.cpp
+
+    \begin{inout}
+        5 \\
+        10 20 30 -3 7 1
+        \tcblower
+        1 \\ 7 \\ -3 \\ 30 \\ 20 \\ 10 \\
+    \end{inout}
+
+\end{examples}
+
+Da je postopek za obračanje seznamov res pravilen, lahko razmislite sami, named
+tega primera je, da vidimo, kako se pri funkcijah dela s seznami. Seznam, ki ga
+želimo obdelati prejmemo kot parameter in ga potem spremenimo, ničesar pa ne
+vrnemo.
+
+
+\section{Rekurzija}
+
+\emph{Rekurzija} je način pisanja programa na način, da funkcija kliče samo
+sebe. Pogosto lahko zanko nadomestimo z rekurzijo, če nam je tak način
+programiranja ljubši.
+
+Recimo, da želimo sešteti vsa števila med 1 in nekim $n$. Z zanko bi to
+naredili tako, da si pripravimo neko srpemenljivko $vsota$ in potem napišemo
```

Comment:
1. _spremenljivko_
2. za spremenljivke uporabljajmo `\verb`. Nasploh v matematičnem načinu ne pisati imen spremenljivk, ki so daljša od ene črke, ker to ne bo obravnavano kot ime ene spremenljivke, pač pa kot produkt večih